zipline/data/data_portal.py

```diff
@@ -1362,3 +1366,38 @@ def get_fetcher_assets(self, day):
                       if isinstance(asset, Asset)]
 
         return asset_list
+
+    def get_simple_transform(self, asset, transform_name, dt, bars=None):
+        if transform_name == "returns":
+            # returns is always calculated over the last 2 days, regardless
+            # of the simulation's data frequency.
+            hst = self.get_history_window(
+                [asset], dt, 2, "1d", "price", ffill=True
+            )[asset]
+
+            return (hst.iloc[-1] - hst.iloc[0]) / hst.iloc[0]
+
+        if bars is None:
+            raise ValueError("bars cannot be None!")
+
+        price_arr = self.get_history_window(
+            [asset], dt, bars, "1m", "price", ffill=True
```

Comment:
Is this always `1m`? i.e. is this only to support Q minute mode, or should we provide this backwards compatibility for public daily mode as well?